application-analytics-test/application-analytics-test-docker/src/test/it/xwiki/analytics/test/ui/AnalyticsIT.java

```diff
@@ -0,0 +1,224 @@
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+@UITest
+public class AnalyticsIT
+{
+    /**
+     * Creates the Admin user
+     */
+    public void setupUsers(TestUtils testUtils)
+    {
+        testUtils.loginAsSuperAdmin();
+        testUtils.setGlobalRights("XWiki.XWikiAdminGroup", "", "admin", true);
+        testUtils.createAdminUser();
+        testUtils.loginAsAdmin();
+    }
+
+    /**
+     * Start the matomo and sql containers, rename the matomo.js to a random int to force the browser to load the
+     * tracking script without explicit settings and generate a new auth token for matomo to be used in the tests.
+     */
+    public void setupContainers(TestConfiguration testConfiguration) throws Exception
+    {
+        GenericContainer<?> sqlContainer = startDb(testConfiguration);
+        GenericContainer<?> matomoContainer = startMatomo(testConfiguration, sqlContainer);
+        Container.ExecResult result = matomoContainer.execInContainer("sh", "-c",
+            "grep -rl 'matomo.js' /var/www/html/ | xargs -d '\\n' -I {} sed -i 's/matomo.js/36011373.js/g' \"{}\"");
+        matomoContainer.execInContainer("sh", "-c", "mv /var/www/html/matomo.js /var/www/html/36011373.js");
+        Config.MATOMO_AUTH_TOKEN =
+            MatomoViewPage.createToken("http://" + Config.ADDRESS + ":" + matomoContainer.getMappedPort(80));
+    }
+
+    /**
+     * Import the platform.html.head UIExtension point to make the tracking code work in the test environmnet.
+     */
+    private void setupUIExtension(TestUtils testUtils) throws Exception
+    {
+        testUtils.setWikiPreference("meta",
+            "#foreach($uix in $services.uix.getExtensions(\"org.xwiki.platform.html.head\","
+                + " {'sortByParameter' : 'order'}))\n" + "  $services.rendering.render($uix.execute(), 'xhtml/1.0')\n"
+                + "#end");
+    }
+
+    @BeforeAll
+    void setup(TestConfiguration testConfiguration, TestUtils testUtils) throws Exception
+    {
+        setupContainers(testConfiguration);
+        setupUIExtension(testUtils);
+        setupUsers(testUtils);
+        HomePageViewPage.gotoPageHomePage();
+    }
+
+    /**
+     * The function checks that the "Save" button displays the correct messages when the configurations provided by the
+     * users are incorrect.
+     */
+    @Test
+    @Order(1)
+    void checkWrongConfigs(XWikiWebDriver driver) throws InterruptedException
+    {
+        AdminViewPage adminViewPage = new AdminViewPage();
+        adminViewPage.gotoAdminPage();
+        System.out.println("/q/q " + Config.MATOMO_AUTH_TOKEN + " /q/q");
+        adminViewPage.setTrackingCode("").setAuthTokenId(Config.MATOMO_AUTH_TOKEN).setIdSiteId("1")
+            .setRequestAddressId(Config.ADDRESS + ":" + Config.MATOMO_BRIDGE_PORT).bringSaveButtonIntoView();
+
+        adminViewPage.inProgressNotification("Saving...");
```

Comment:
Why not use directly `adminViewPage.waitForNotificationInProgressMessage("Saving...");` ? The name is also more suggestive